hal/src/rtl872x/deviceid_hal.cpp

```diff
@@ -92,18 +99,29 @@ unsigned hal_get_device_id(uint8_t* dest, unsigned destLen)
     return HAL_DEVICE_ID_SIZE;
 }
 
-unsigned hal_get_platform_id()
-{
+unsigned hal_get_platform_id() {
     return PLATFORM_ID;
 }
 
-int HAL_Get_Device_Identifier(const char** name, char* buf, size_t buflen, unsigned index, void* reserved)
-{
-    return -1;
+int HAL_Get_Device_Identifier(const char** name, char* buf, size_t buflen, unsigned index, void* reserved) {
+    return SYSTEM_ERROR_NOT_SUPPORTED;
+}
+
+int hal_get_ble_mac_address(uint8_t* dest, size_t destLen, void* reserved) {
```

Comment:
Mind generalizing this API and adding MAC index definitions into implementation header? See https://github.com/particle-iot/device-os/pull/2492#discussion_r944431649 for suggestions. Let's also convert normal big-endian MAC to BLE expected format inside BLE hal as opposed to here.